### PR TITLE
React 0.13 Update

### PR DESCRIPTION
**Pull request by malonecj**
- [x] Upgrade React
 - [x] Fix issues due to breaking changes
 - [x] Es6 class editors
    - [x] Editor Base class
    - [x] SimpleTextEditor extending EditorBase
    - [x] DropDownEditor extending EditorBase 
    - [x] Autocomplete editor extending EditorBase 
    - [x] Editor container tests
  
react-data-grid

### CODE REVIEW

File: src/addons/editors/SimpleTextEditor.js [hunk 1]
```diff
@@ -9,30 +9,15 @@
...
-
-  getInputNode(): HTMLInputElement{
-    return this.getDOMNode();
-  },
+class SimpleTextEditor extends EditorBase {
 
   render(): ?ReactElement {
-    return (<input ref="input" type="text" onBlur={this.props.onBlur} className="form-control" defaultValue={this.props.value} onKeyDown={this.props.onKeyDown} />);
+    return (<input type="text" className="form-control" defaultValue={this.props.value} />);
   }
```
Comment (malonecj) [inline src/addons/editors/SimpleTextEditor.js]: @bakesteve, got bored in Ireland over Easter. Have simplified the editors. No longer have to worry about binding onKeyDown as this is taken care of in EditorContainer unless you want to override default behaviour

File: themes/react-data-grid.css [hunk 2]
```diff
@@ -295,105 +295,232 @@ input.react-grid-CheckBox {
...
+  max-width: 100%;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
+.has-value > .Select-control > .Select-placeholder {
+  color: #333333;
+}
+.Select-input > input {
+  cursor: default;
```
Comment (malonecj) [inline themes/react-data-grid.css]: Should probably remove this from CSS file, and say that React-Select is a dependency that needs to be included for react-data-grid-with-addons

File: src/addons/editors/SimpleTextEditor.js (hunk 1 above)
Comment (bakesteve) [inline src/addons/editors/SimpleTextEditor.js]: :+1: (to the work, not getting bored…!)

File: themes/react-data-grid.css (hunk 2 above)
Comment (bakesteve) [inline themes/react-data-grid.css]: Yep

Seems in react-think css is part of your code?

I’d probably add a nuance to that, in that you can have base css (ie assume bootstrap) in the same way you assume es5 or es6) but then any very specific css comes with the component (need to think how we manage that across components though, and how we bundle the assets)

**Comment by Blystad** [comment]
Any chance this can/will be merged/finished soon?

**Comment by maiis** [comment]
:+1:

**Comment by malonecj** [comment]
I've been a bit stuck for time at the moment. But have not forgotten about this. It is practically finished. There are still some failing tests and the autocomplete edtior needs to be complete. Hope to get around to it soon

**Comment by jbrodriguez** [comment]
Hi @malonecj, nice library !

Can you ship a pre/alpha/beta without autocomplete for the time being ?

I tried installing with `npm install git://github.com/adazzle/react-data-grid.git#cm-react013-upgrade`, but it fails saying it didn't find `./dist/react-data-grid`.

The dist folder isn't there.

**Comment by malonecj** [comment]
Hi yes

This will be on its ways in the next couple of days.

Thanks

**Comment by aidanlister** [comment]
Exciting

**Comment by malonecj** [comment]
Hi all

I've published a pre release version 0.13.0-alpha1

The autocomplete is working, but there are still two failing tests at the moment. I'll try to pick those up soon or feel free anyone to step in.

**Comment by malonecj** [comment]
@bakesteve Build passing along with all tests now.

**Pull request closed without merging**